akvo/rsr/tests/rest/test_permissions.py

```diff
@@ -0,0 +1,450 @@
+# -*- coding: utf-8 -*-
+
+"""
+Akvo RSR is covered by the GNU Affero General Public License.
+
+See more details in the license.txt file located at the root folder of the Akvo RSR module.
+For additional details on the GNU license please see < http://www.gnu.org/licenses/agpl.html >.
+"""
+
+import re
+
+from django.conf import settings
+from django.contrib.auth import get_user_model
+from django.contrib.auth.models import AnonymousUser, Group
+from django.core import management
+from django.test import TestCase
+
+from akvo.rest.viewsets import PublicProjectViewSet
+from akvo.rsr import models as M
+from akvo.utils import check_auth_groups
+
+
+def slugify(s):
+    return re.sub('\W+', '-', s.lower())
+
+
+User = get_user_model()
+
+
```

Comment:
This is just because I don't like anonymous tuples 😉... 

The counts tuples could be made into namedtuples:
```python
import collections
# gc is the number of objects visible to that group, groupcount
gc = collections.namedtuple('gc', 'admin anonymous editor other')
...
            model_map[M.Project] = {'counts': gc(8, 4, 6, 4), 'project_relation': ''}
```
etc.

Then iter_queryset() can be nicely shortened:

```python
def iter_queryset(self, user_type='other'):
    """An iterator over querysets to test permissions with."""

    for model in self.model_map:
        data = self.model_map[model]
        count = getattr(data['counts'], user_type)
        yield model.objects.all(), data['project_relation'], count
```